ambry-clustermap/src/main/java/com.github.ambry.clustermap/HelixClusterManager.java

```diff
@@ -153,6 +153,9 @@ public HelixClusterManager(ClusterMapConfig clusterMapConfig, String instanceNam
               manager.connect();
               logger.info("Established connection to Helix manager at {}", zkConnectStr);
             }
+
+            CountDownLatch routingTableInitLatch = new CountDownLatch(1);
```

Comment:
I would recommend instantiating the `CountDownLatch` within `ClusterChangeHandler` since instantiating it outside leaks some internal details, like the count you are waiting for. Maybe add a method like `CountDownLatch::waitForStartup(int minutesToWait)`

Reply:
I guess you are referring to `ClusterChangeHandler.waitForStartup(int waitTimeout)`? (Correct me if I am wrong)